### PR TITLE
Show notification for poll start events but not for responses

### LINKED ISSUE #4780: Poll system notifications on Android are not user friendly
### Steps to reproduce

1. Have a poll in a room with noisy notifications
2. Receive system notification when someone votes

#### What did you expect?
A message saying "X voted in poll Y" or something similar

#### What happened instead?

Screenshot_20211220-171830

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

1.3.10

### Homeserver

_No response_

### Will you send logs?

No

### PR DESCRIPTION
**Pull request by onurays**
Fixes #4780 

- We still need to set push rules for `org.matrix.msc3381.poll.start` for clear rooms
- Poll responses will not anymore update room summary item in room list
- We rely on badge count from sync response, so unread count continues to increase for poll responses

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/notifications/NotifiableEventResolver.kt [hunk 1]
```diff
@@ -94,7 +94,7 @@ class NotifiableEventResolver @Inject constructor(
     }
 
     suspend fun resolveInMemoryEvent(session: Session, event: Event, canBeReplaced: Boolean): NotifiableEvent? {
-        if (event.getClearType() != EventType.MESSAGE) return null
+        if (event.getClearType() !in listOf(EventType.MESSAGE, EventType.POLL_START)) return null
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/features/notifications/NotifiableEventResolver.kt:97]: I'm wondering if we could create an extension for this check and reuse in both places, could make adding more types easier in the future :crossed_fingers: 

```kotlin
fun Event.supportsNotification() = this.clearType in listOf(EventType.MESSAGE, EventType.POLL_START)
```

what do you think?

**Review by ouchadam — APPROVED** [review]
LGTM :+1:  small comment about extracting an extension but not a blocker

File: vector/src/main/java/im/vector/app/features/notifications/NotifiableEventResolver.kt [hunk 2]
```diff
@@ -153,7 +153,8 @@ class NotifiableEventResolver @Inject constructor(
             event.attemptToDecryptIfNeeded(session)
             // only convert encrypted messages to NotifiableMessageEvents
             when (event.root.getClearType()) {
-                EventType.MESSAGE -> {
+                EventType.MESSAGE,
+                EventType.POLL_START -> {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/notifications/NotifiableEventResolver.kt:157]: will be replaced by `in EventType.POLL_START` once #5345 will be merged (there is just a minor fix to do there)

**Review by bmarty — APPROVED** [review]
LGTM, just a small reminder :)

File: vector/src/main/java/im/vector/app/features/notifications/NotifiableEventResolver.kt (hunk 1 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/notifications/NotifiableEventResolver.kt]: Nice idea, done.

**Pull request merged**